drivers/can/can_stm32.c

```diff
@@ -526,10 +525,10 @@ int can_stm32_recover(struct device *dev, s32_t timeout)
 
 	can_leave_init_mode(can);
 
-	start_time = k_cycle_get_32();
+	start_time = k_uptime_ticks();
 
 	while (can->ESR & CAN_ESR_BOFF) {
-		if (k_cycle_get_32() - start_time >= CAN_BOFF_RECOVER_TIMEOUT) {
+		if (k_uptime_ticks() - start_time >= timeout.ticks) {
```

Comment:
Does this not need to check for the special value K_FOREVER?

Reply:
You are right. Good catch!